src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenTupleTest.cs

```diff
@@ -1527,5 +1598,360 @@ static void Main()
+    {
+        System.Console.WriteLine(""third"");
+        System.Console.WriteLine(x.Item1().ToString());
+    }
+}
+" + trivial2uple + trivial3uple + trivalRemainingTuples;
+
+            var comp = CompileAndVerify(source, expectedOutput: @"
+second
+first
+third
+7
+");
+        }
+
+        [Fact]
+        public void TargetTypingNullable02()
+        {
+            var source = @"
+using System;
+class C
+{
+    static void Main()
+    {
+        var x = M1();
+        Test(x);
+    }
+
+    static (int a, string b)? M1()
+    {
+        return (1, null);
+    }
+
+    static void Test<T>(T arg)
+    {
+        System.Console.WriteLine(typeof(T));
+        System.Console.WriteLine(arg);
+    }
+}
+" + trivial2uple;
+
+            var comp = CompileAndVerify(source, expectedOutput: @"
+System.Nullable`1[System.ValueTuple`2[System.Int32,System.String]]
+{1, }
+");
+        }
+
+        [Fact]
+        public void TargetTypingNullable02Long()
+        {
+            var source = @"
+using System;
+class C
+{
+    static void Main()
+    {
+        var x = M1();
+        System.Console.WriteLine(x?.a);
+        System.Console.WriteLine(x?.a8);
+        Test(x);
+    }
+
+    static (int a, string b, int a1, int a2, int a3, int a4, int a5, int a6, int a7, int a8)? M1()
+    {
+        return (1, null, 1, 2, 3, 4, 5, 6, 7, 8);
+    }
+
+    static void Test<T>(T arg)
+    {
+        System.Console.WriteLine(arg);
+    }
+}
+" + trivial2uple + trivial3uple + trivalRemainingTuples;
+
+            var comp = CompileAndVerify(source, expectedOutput: @"
+1
+8
+System.ValueTuple`8[System.Int32,System.String,System.Int32,System.Int32,System.Int32,System.Int32,System.Int32,System.ValueTuple`3[System.Int32,System.Int32,System.Int32]]
+");
+        }
+
+        [Fact]
+        public void TargetTypingNullableOverload()
+        {
+            var source = @"
+using System;
+class C
+{
+    static void Main()
+    {
+        Test((null, null, null, null, null, null, null, null, null, null));
+        Test((""a"", ""a"", ""a"", ""a"", ""a"", ""a"", ""a"", ""a"", ""a"", ""a""));
+        Test((1, 1, 1, 1, 1, 1, 1, 1, 1, 1));
+    }
+
+    static void Test((string, string, string, string, string, string, string, string, string, string) x)
+    {
+        System.Console.WriteLine(""first"");
+    }
+
+    static void Test((string, string, string, string, string, string, string, string, string, string)? x)
+    {
+        System.Console.WriteLine(""second"");
+    }
+
+    static void Test((int, int, int, int, int, int, int, int, int, int)? x)
+    {
+        System.Console.WriteLine(""third"");
+    }
+
+    static void Test((int, int, int, int, int, int, int, int, int, int) x)
+    {
+        System.Console.WriteLine(""fourth"");
+    }
+}
+" + trivial2uple + trivial3uple + trivalRemainingTuples;
+
+            var comp = CompileAndVerify(source, expectedOutput: @"
+first
+first
+fourth
+");
+        }
```

Comment:
Do you want a test for `var x = (null, null);` as well (no target type and no natural type)?